neuralprophet/forecaster.py

```diff
@@ -1711,8 +1735,412 @@ def plot_latest_forecast(
                 line_per_origin=True,
             )
 
+    def check_if_configured(self, components, error_flag=False):
+        """Check if components were set in the model configuration by the user.
+
+        Parameters
+        ----------
+            components : str or list, optional
+                name or list of names of components to check
+
+                Options
+                ----
+                * ``trend``
+                * ``trend_rate_change``
+                * ``seasonality``
+                * ``autoregression``
+                * ``lagged_regressors```
+                * ``events``
+                * ``future_regressors`
+                * ``residuals``
+                * ``uncertainty``
+            error_flag : bool
+                Activate to raise a ValueError if component has not been configured
+
+        Returns
+        -------
+            components
+                list of components only including the components set in the model configuration
+        """
+        # Error if selected plotting components are not set in the model configuration by the user
+        for comp in reversed(components):
```

Comment:
If I understood the implementation right, this seems a bit inefficient - we validate all components for each component. Wouldn't it be better to only validate each component once (independent of the others)?

Reply:
Also why was this moved into `forecaster.py`? To me this is all purely plotting related and would better remain in `plot_forecast.py`

Reply:
Definitely true ^^ will change